src/main/java/com/box/sdk/MetadataTemplate.java

```diff
@@ -309,6 +314,55 @@ public static void deleteMetadataTemplate(BoxAPIConnection api, String scope, St
         request.send();
     }
 
+    public static BoxResourceIterable<BoxItem.Info> executeMetadataQuery(final BoxAPIConnection api,
+                                            String from, String query, JsonObject queryParameters,
+                                            String ancestorFolderId, String indexName,
+                                            JsonArray orderBy, int limit, String marker) {
+
+        JsonObject jsonObject = new JsonObject().add("from", from);
+        if (query != null) {jsonObject.add("query", query);}
+        if (queryParameters != null) {jsonObject.add("query_params", queryParameters);}
+        if (ancestorFolderId != null) {jsonObject.add("ancestor_folder_id", ancestorFolderId);}
+        if (indexName != null) {jsonObject.add("use_index", indexName);}
+        if (orderBy != null) {jsonObject.add("order_by", orderBy);}
+        jsonObject.add("limit", limit);
+        if (marker != null) {jsonObject.add("marker", marker);}
```

Comment:
I don't think you need this line now.
```suggestion
```

Reply:
Yeah, the `metadata_queries/execute_read` doesn't take ANY query parameters.  Any passed in are ignored.  In order for for the marker to be respected, it has to be sent in the body.

Reply:
Then the iterator has to be updated to send the marker in the body rather than the query as it is doing now.

Reply:
It's doing both.  jsonObject is the body.  It gets passed in there and then again in the marker parameter.
I chose not to hard-code a null for the marker parameter so that any future changes will be supported.  (It's currently ignored).

Reply:
For the first request the marker is passed. For subsequent requests using the iterator, the next marker won't be passed in the body.

Reply:
Good call!  Done!